sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -285,4 +289,52 @@ export class EducationProgramOfferingController {
       studyStartDate: offering.studyStartDate,
     };
   }
+
+  /**
+   * Get programs for a particular institution in paginated.
+   * @param institutionId id of the institution.
+   * @param take is the number of rows shown in the table
+   * @param skip is the number of rows that is skipped/offset from the total list.
+   * For example page 2 the skip would be 10 when we select 10 rows per page.
+   * @param dateSubmittedOrder the sorting order of the submitted date, default its ascending
+   * @param searchProgramName Search the program name in the query
+   * @returns ProgramsOfferingSummaryPaginated.
+   */
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Groups(UserGroups.AESTUser)
+  @Get("institution/:institutionId/programs/take/:take/skip/:skip")
+  async getPaginatedProgramsForInstitution(
+    @Param("institutionId") institutionId: number,
+    @Param("take") take: number,
+    @Param("skip") skip: number,
+    @Query("dateSubmittedOrder") dateSubmittedOrder: number,
```

Comment:
I was assuming that this would be more flexible in the sense that it would be receiving the "column" that it would be ordered.

Reply:
As only single column is used for the ordering, did not want to use another parameter to pass the column

Reply:
I didn't get this part. I believe, we need to sort by date righT?

Reply:
The event from the UI is returning us -1 or 1 for the sort order, so I am manipulating them inside the controller as ASC and DESC.

Reply:
I think it's better to manipulate it in web, what if in the future when we implement vuetify and the sort return is something else?